src/wallet/scriptpubkeyman.h

```diff
@@ -0,0 +1,479 @@
+        :   IsWalletFlagSet(is_set_func),
+            SetWalletFlag(set_flag_func),
+            UnsetWalletFlagWithDB(unset_flag_func),
+            CanSupportFeature(feature_sup_func),
+            GetDisplayName(wallet_name_func),
+            SetMinVersion(set_version_func),
+            m_database(database)
+        {}
+
+    virtual ~ScriptPubKeyMan() {};
+    virtual bool GetNewDestination(const OutputType type, CTxDestination& dest, std::string& error) { return false; }
+    virtual isminetype IsMine(const CScript& script) const { return ISMINE_NO; }
+
+    virtual bool IsCrypted() const { return false; }
+    virtual bool IsLocked() const { return false; }
+    virtual bool Lock() { return false; }
+
+    virtual bool Unlock(const CKeyingMaterial& vMasterKeyIn, bool accept_no_keys = false) { return false; }
+    virtual bool Encrypt(CKeyingMaterial& vMasterKeyIn, WalletBatch* batch) { return false; }
+
+    virtual bool GetReservedDestination(const OutputType type, bool internal, CTxDestination& address, int64_t& index, CKeyPool& keypool) { return false; }
+    virtual void KeepDestination(int64_t index) {}
+    virtual void ReturnDestination(int64_t index, bool internal, const CTxDestination& addr) {}
+
+    virtual bool TopUp(unsigned int size = 0) { return false; }
+
+    //! Mark unused addresses as being used
+    virtual void MarkUnusedAddresses(const CScript& script) {}
+
+    //! Upgrade stored CKeyMetadata objects to store key origin info as KeyOriginInfo
+    virtual void UpgradeKeyMetadata() {}
+
+    /** Sets up the key generation stuff.
+      * Returns false if already setup or setup fails, true if setup is successful
+      * Set force=true to make it re-setup if already setup, used for upgrades
+      */
+    virtual bool SetupGeneration(bool force = false) { return false; }
+
+    /* Returns true if HD is enabled */
+    virtual bool IsHDEnabled() const { return false; }
+
+    /* Returns true if the wallet can give out new addresses. This means it has keys in the keypool or can generate new keys */
+    virtual bool CanGetAddresses(bool internal = false) { return false; }
+
+    /** Upgrades the wallet to the specified version */
+    virtual bool Upgrade(int prev_version, int new_version, std::string& error) { return false; }
+
+    virtual bool HavePrivateKeys() const { return false; }
+
+    //! The action to do when the DB needs rewrite
+    virtual void RewriteDB() {}
+
+    virtual int64_t GetOldestKeyPoolTime() { return GetTime(); }
+
+    virtual size_t KeypoolCountExternalKeys() { return 0; }
+    virtual unsigned int GetKeypoolSize() const { return 0; }
+
+    virtual int64_t GetTimeFirstKey() const { return 0; }
+
+    virtual const CKeyMetadata* GetMetadata(uint160 id) const { return nullptr; }
```

Comment:
Brief comment on what this does? (I don't think similar existed before in CWallet)

Reply:
It just returns a `CKeyMetadata` as the function name suggests.

It is needed so that getting metadata no longer requires direct access to `mapKeyMetadata` which no longer exists in CWallet and is not guaranteed to exist in other ScriptPubKeyMans (e.g. it is generated on the fly in native descriptor wallets).